app/frontend/components/pages/sendAda/confirmTransactionDialog.tsx

```diff
@@ -254,65 +254,93 @@ const ConvertFundsReview = ({
 }
 
 const ConfirmTransactionDialog = () => {
-  const {rawTransactionOpen, transactionSummary, txConfirmType} = useSelector((state) => ({
+  const {
+    rawTransactionOpen,
+    transactionSummary,
+    txConfirmType,
+    isCrossAccount,
+    cachedTransactionSummaries,
+    sendAddress,
+    sourceAccountIndex,
+  } = useSelector((state) => ({
     transactionSummary: state.sendTransactionSummary,
     rawTransactionOpen: state.rawTransactionOpen,
     txConfirmType: state.txConfirmType,
+    isCrossAccount: state.isCrossAccount,
+    cachedTransactionSummaries: state.cachedTransactionSummaries,
+    sendAddress: state.sendAddress.fieldValue,
+    sourceAccountIndex: state.sourceAccountIndex,
   }))
   const {setRawTransactionOpen, submitTransaction, cancelTransaction} = useActions(actions)
 
+  const hideDefaultSummary = txConfirmType === TxType.DEREGISTER_STAKE_KEY
+
   const titleMap: {[key in TxType]: string} = {
     [TxType.DELEGATE]: 'Delegation review',
     [TxType.SEND_ADA]: 'Transaction review',
     [TxType.CONVERT_LEGACY]: 'Stakable balance conversion review',
     [TxType.WITHDRAW]: 'Rewards withdrawal review',
     [TxType.POOL_REG_OWNER]: '',
     [TxType.DEREGISTER_STAKE_KEY]: 'Deregister stake key',
-    // crossAccount: 'Transaction between accounts review',
   }
-  const hideDefaultSummary = transactionSummary.type === TxType.DEREGISTER_STAKE_KEY
-  // TODO: refactor, remove txConfirmType
+
+  const onSubmit = () => {
+    // refactored cases
+    if ([TxType.DEREGISTER_STAKE_KEY].includes(txConfirmType)) {
+      return submitTransaction({
+        sendAddress,
+        sourceAccountIndex,
+        txSummary: cachedTransactionSummaries[TxType.DEREGISTER_STAKE_KEY],
+      })
+    }
+    // to be refactored cases
+    return submitTransaction({sendAddress, sourceAccountIndex, txSummary: transactionSummary})
+  }
+
+  // Refactor: tmp util till all transactions types use "cachedTransactionSummaries"
+  const getModalBody = () => {
```

Comment:
I did not want to "refactor" all, as it would be a nightmare to review to I decided to have it split like this.

Reply:
Now it doesn't make sense, but I'd later opt for handling all txs types here (now, TxType.POOL_REG_OWNER is handled separately) and when that's done, I would take advantage of typescript's exhaustive switches and our "assertUnreachable". If you agree, you can comment it down or I'll create an issue. 

A switch would look nicer than "ifs" even now imho

Reply:
Actually, we would probably opt for having multiple smaller and specific dialogs as discussed so I am not sure whether comment it out inside this PR. But I can at least change it to `switch`.